lib/diagnostic.js

```diff
@@ -10,26 +10,41 @@ const normalizeWords = function (words) {
   let wordArray = words.split(' ');
   const normalize = function (word) {
     return word.replace(/\W+/,'').toUpperCase();
-  }
+  };
 return wordArray.map(normalize);
 };
-
 // 2. Write a function that takes a string argument
 // and returns an array of unique normalized words.
-const uniqueWords = function (/* */) {
-  /* your response here */
-};
+const uniqueWords = function (array) {
+  let result = {};
+  array.forEach(function (word) {
+    result[word] = 0;
+  });
+    //take each value in the array
+  return result;};
 
 // 3. Write a function that returns the count of words in a string.
-// Provid the *option* to count unique words instead of total words.
-const wordCount = function (/* */) {
-  /* your response here */
+// Provide the *option* to count unique words instead of total words.
+const wordCount = function (stringOfWords, countUnique) {
+  if(countUnique) {
+    let arrayOfWords = normalizeWords(stringOfWords);
+    let uniqueCount = uniqueWords(arrayOfWords);
+    return uniqueCount.length;
+
+  }
 };
 
```

Comment:
This looks super close, but returns undefined regardless of what we put in as countUnique. There is also no 'else' statement. My guess is if you had to do this now you wouldn't have any issues at all...